contracts/truefi2/TrueFiPool2.sol

```diff
@@ -0,0 +1,414 @@
+     * @param staker Account exiting
+     * @param amount Amount unstaking
+     */
+    event Exited(address indexed staker, uint256 amount);
+
+    /**
+     * @dev Emitted when funds are flushed into curve.fi
+     * @param currencyAmount Amount of currencyTokens deposited
+     */
+    event Flushed(uint256 currencyAmount);
+
+    /**
+     * @dev Emitted when funds are pulled from curve.fi
+     * @param yAmount Amount of pool tokens
+     */
+    event Pulled(uint256 yAmount);
+
+    /**
+     * @dev Emitted when funds are borrowed from pool
+     * @param borrower Borrower address
+     * @param amount Amount of funds borrowed from pool
+     * @param fee Fees collected from this transaction
+     */
+    event Borrow(address borrower, uint256 amount, uint256 fee);
+
+    /**
+     * @dev Emitted when borrower repays the pool
+     * @param payer Address of borrower
+     * @param amount Amount repaid
+     */
+    event Repaid(address indexed payer, uint256 amount);
+
+    /**
+     * @dev Emitted when fees are collected
+     * @param beneficiary Account to receive fees
+     * @param amount Amount of fees collected
+     */
+    event Collected(address indexed beneficiary, uint256 amount);
+
+    /**
+     * @dev Emitted when joining is paused or unpaused
+     * @param isJoiningPaused New pausing status
+     */
+    event JoiningPauseStatusChanged(bool isJoiningPaused);
+
+    /**
+     * @dev only lender can perform borrowing or repaying
+     */
+    modifier onlyLender() {
+        require(msg.sender == address(lender), "TrueFiPool: Caller is not the lender");
+        _;
+    }
+
+    /**
+     * @dev pool can only be joined when it's unpaused
+     */
+    modifier joiningNotPaused() {
+        require(!isJoiningPaused, "TrueFiPool: Joining the pool is paused");
+        _;
+    }
+
+    /**
+     * @dev Allow pausing of deposits in case of emergency
+     * @param status New deposit status
+     */
+    function changeJoiningPauseStatus(bool status) external onlyOwner {
+        isJoiningPaused = status;
+        emit JoiningPauseStatusChanged(status);
+    }
+
+    /**
+     * @dev Virtual value of liquid assets in the pool
+     * @return Virtual liquid value of pool assets
+     */
+    function liquidValue() public view returns (uint256) {
+        return currencyBalance().add(strategy.value());
+    }
+
+    /**
+     * @dev Calculate pool value in TUSD
+     * "virtual price" of entire pool - LoanTokens, TUSD, curve y pool tokens
+     * @return pool value in USD
+     */
+    function poolValue() public view returns (uint256) {
+        // this assumes defaulted loans are worth their full value
+        return liquidValue().add(loansValue());
+    }
+
+    /**
+     * @dev Virtual value of loan assets in the pool
+     * Will return cached value if inSync
+     * @return Value of loans in pool
+     */
+    function loansValue() public view returns (uint256) {
+        return lender.value();
```

Comment:
this function will no longer work unless we have 1 lender for each pool.
Let's discuss this Monday, whether we need multiple lenders or a single lender.

Reply:
In this PR we only extracted strategy from TrueFiPool and deployed it with factory. This line will be changed when we'll implement LenderV2